src/Thermochimica-cxx.C

```diff
@@ -0,0 +1,324 @@
+  {
+    double molesPhase;
+    int info;
+    TCAPI_getPureConPhaseMol(phaseName.c_str(), phaseName.length(), &molesPhase, &info);
+    return {molesPhase, info};
+  }
+
+  std::pair<int, int>
+  getPhaseIndex(const std::string &phaseName)
+  {
+    int index, info;
+    TCAPI_getPhaseIndex(phaseName.c_str(), phaseName.length(), &index, &info);
+    return {index, info};
+  }
+
+  std::pair<double, int>
+  getOutputSiteFraction(const std::string &phaseName, int sublattice, int constituent)
+  {
+    double siteFraction;
+    int info;
+    TCAPI_getOutputSiteFraction(phaseName.c_str(), phaseName.length(), &sublattice, &constituent, &siteFraction, &info);
+    return {siteFraction, info};
+  }
+
+  std::pair<double, int>
+  getSublSiteMol(const std::string &phaseName, int sublattice, int constituent)
+  {
+    double siteMoles;
+    int info;
+    TCAPI_getSublSiteMol(phaseName.c_str(), phaseName.length(), &sublattice, &constituent, &siteMoles, &info);
+    return {siteMoles, info};
+  }
+
+  // MQMQA functions
+
+  std::pair<double, int>
+  getMqmqaMolesPairs(const std::string &phaseName)
+  {
+    double molesPairs;
+    int info;
+    TCAPI_getMqmqaMolesPairs(phaseName.c_str(), phaseName.length(), &molesPairs, &info);
+    return {molesPairs, info};
+  }
+
+  std::pair<double, int>
+  getMqmqaPairMolFraction(const std::string &phaseName, const std::string &pairName)
+  {
+    double molesPairs;
+    int info;
+    TCAPI_getMqmqaPairMolFraction(phaseName.c_str(), phaseName.length(), pairName.c_str(), pairName.length(), &molesPairs, &info);
+    return {molesPairs, info};
+  }
+
+  std::tuple<int, int, int>
+  getMqmqaNumberPairsQuads(const std::string &phaseName)
+  {
+    int nPairs;
+    int nQuads;
+    int info;
+    TCAPI_getMqmqaNumberPairsQuads(phaseName.c_str(), phaseName.length(), &nPairs, &nQuads, &info);
+    return {nPairs, nQuads, info};
+  }
+
+  std::pair<double, int>
+  getMqmqaConstituentFraction(const std::string &phaseName, int sublattice, const std::string &constituent)
+  {
+    double moleFraction;
+    int info;
+    TCAPI_getMqmqaConstituentFraction(phaseName.c_str(), phaseName.length(), &sublattice, constituent.c_str(), constituent.length(), &moleFraction, &info);
+    return {moleFraction, info};
+  }
+
+  reinitData getReinitData()
+  {
+    reinitData data;
+    int available;
+    auto [elements, species] = getReinitDataSizes();
+    data.assemblage.resize(elements);
+    data.molesPhase.resize(elements);
+    data.elementPotential.resize(elements);
+    data.chemicalPotential.resize(species);
+    data.moleFraction.resize(species);
+
+    TCAPI_getReinitData(data.assemblage.data(), data.molesPhase.data(), data.elementPotential.data(), data.chemicalPotential.data(), data.moleFraction.data(), data.elementsUsed.data(), &available);
+
+    data.reinitAvailable = false;
+    if (available > 0)
+    {
+      data.reinitAvailable = true;
+    }
+
+    return data;
+  }
+
+  void setReinitData(reinitData data)
```

Comment:
See if 

```suggestion
  void setReinitData(const reinitData & data)
```

work. Or if you have to drop the `const`. But either way you should pass this by reference.

Reply:
Good idea. `const` works, as long as it's consistent all the way through.